SharedControls/Viewers/BuildingViewer.xaml.cs

```diff
@@ -15,41 +15,63 @@ public partial class BuildingViewer : Window, IAssignmentViewer
         public event Action<IAssignment, bool> CompletionStateChanged;
 
         private readonly BuildingAssignment _assignment;
+        private int _currentItemIndex = 0;
 
         public string TaskTitle { get; }
+
         public BuildingViewer(BuildingAssignment assignment)
         {
             InitializeComponent();
             DataContext = this;
 
             _assignment = assignment;
-
             TaskTitle = _assignment.Title;
 
-            foreach (var item in _assignment.Items)
+            // Загрузка первого выражения
+            LoadCurrentItem();
+        }
+
+        // Метод для загрузки текущего элемента
+        private void LoadCurrentItem()
+        {
+            spItems.Children.Clear();
+
+            if (_currentItemIndex < _assignment.Items.Count)
```

Comment:
this comparison is unnecessary, LoadCurrentItem will never be called if currentItemIndex isn't less than Items.Count